### DISCUSSION TITLE
Does Codemagic support waiting for other workflows to finish?

### DISCUSSION THREAD

**Discussion by incendial** [codemagic-ci-cd/codemagic-docs#1235] [Q&A]
Hi, my use-case is to compile a Dart application and (since [Dart doesn't support cross-compilation](https://github.com/dart-lang/sdk/issues/28617)) I need a separate workflow for each target platform to compile. So I'd like to wait for all these workflows to complete and then create a GitHub release with the artefacts from then. Is it possible? If waiting is not possible, any guidance how to achieve the expected result (like having 1 GitHub release with all the artefacts)? 

And does Codemagic support matrix set up like [GitHub actions do](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs)?

**Comment by mikhail-tokarev** [comment]
Hi, 
This is a nice feature request since currently we don't have matrix support or any similar out-of-the-box functionality.

Internally we use some workaround, so hopefully you can adopt it for your case too:
1. we trigger parallel workflows with webhook or rest api
2. at the end each build updates its status on shared storage (more about it below)
3. after that it checks if all workflows updated their statuses or not. If all statuses are updated it triggers the final workflow with rest api
4. if build is not the last one, no extra actions required.

Depends on the case it can be different shared storages: 
- for instance some bucket (S3 or Google storage) where you upload file and check how many files already uploaded to understand if it's final one or not
- for GitHub it can be Pull Request labels (very easy to add/list with GitHub CLI).

I hope it will help

**Reply by incendial** [reply]
> This is a nice feature request since currently we don't have matrix support or any similar out-of-the-box functionality.

Sad to hear that! Thank you for the suggestion (I'll try something similar to labels, I think), but having real matrix support would be amazing!
